File: backend/app/schemas/remote_plugin.py
"""远程插件 Pydantic schemas。"""
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class RemotePluginCreate(BaseModel):
    source_url: str
    default_enabled: bool = False


class RemotePluginOut(BaseModel):
    id: int
    name: str
    display_name: str
    description: str
    author: str
    source_url: str
    version: str
    enabled: bool
    default_enabled: bool = False
    installed_at: datetime | None = None

    class Config:
        from_attributes = True


class RegistryPluginOut(BaseModel):
    name: str
    display_name: str
    description: str
    author: str
    source_url: str
    version: str
    installed: bool

File: frontend/src/types/remotePlugin.ts
export interface RemotePlugin {
  id: number;
  name: string;
  display_name: string;
  description: string;
  author: string;
  source_url: string;
  version: string;
  enabled: boolean;
  default_enabled: boolean;
  installed_at: string | null;
}

export interface InstallRequest {
  source_url: string;
  default_enabled?: boolean;
}

export interface AccountPluginAction {
  account_ids: number[];
}
